include/xtensor/xcontainer.hpp

```diff
@@ -753,12 +644,21 @@ namespace xt
      * Reshapes the container.
      * @param shape the new shape
      */
-    template <class D>
-    inline void xstrided_container<D>::reshape(const shape_type& shape)
+    template <class D, class L>
+    inline void xstrided_container<D, L>::reshape(const shape_type& shape)
     {
         if (shape != m_shape)
         {
-            reshape(shape, layout::row_major);
+            if (std::is_same<L, xt::layout_type::row_major>::value)
+            {
+                std::cout << "RESHAPING WITH ROW_MAJOR" << std::endl;
+                reshape(shape, layout::row_major);
+            }
+            else
+            {
+                std::cout << "RESHAPING WITH COL_MAJOR" << std::endl;
+                reshape(shape, layout::column_major);
+            }
```

Comment:
I guess that we should default to `row_major`, instead of defaulting to `column_major` when the layout type is custom.